docs/shared/recipes/tips-n-tricks/switch-to-workspaces-project-references.md

```diff
@@ -172,26 +172,79 @@ The root `tsconfig.json` file should extend `tsconfig.base.json` and not include
 {% /tab %}
 {% /tabs %}
 
+## Register Nx Typescript Plugin
+
+Make sure that the `@nx/js` plugin is installed in your repository and `@nx/js/typescript` is registered as a plugin in the `nx.json` file.
+
+```jsonc {% fileName="nx.json" %}
+{
+  "plugins": [
+    {
+      "plugin": "@nx/js/typescript",
+      "options": {
+        "typecheck": {
+          "targetName": "typecheck"
+        },
+        "build": {
+          "targetName": "build",
+          "configName": "tsconfig.lib.json",
+          "buildDepsName": "build-deps",
+          "watchDepsName": "watch-deps"
+        }
+      }
+    }
+  ]
+}
+```
+
+This plugin will register a [sync generator](/concepts/sync-generators) to automatically maintain the project references across the repository.
+
 ## Create Individual Project package.json files
 
 When using package manager project linking, every project needs to have a `package.json` file. You can leave all the task configuration in the existing `project.json` file. For application projects, you only need to specify the `name` property. For library projects, you should add an `exports` property that accounts for any TypeScript path aliases that referenced the project. A typical configuration is shown below:
 
+{% tabs %}
+{% tab label="Non-buildable library" %}
+
+```json {% fileName="libs/ui/package.json" %}
+{
+  "name": "@myorg/ui",
+  "exports": {
+    ".": {
+      "types": "./src/index.d.ts",
+      "import": "./src/index.js",
+      "default": "./src/index.js"
+    }
+  }
+}
+```
+
+{% /tab %}
+{% tab label="Buildable Library" %}
+
 ```json {% fileName="libs/ui/package.json" %}
 {
   "name": "@myorg/ui",
   "exports": {
-    ".": "./src/index.js"
+    ".": {
+      "types": "./src/index.d.ts",
+      "import": "./dist/index.js",
+      "default": "./dist/index.js"
+    }
   }
 }
 ```
 
+{% /tab %}
+{% /tabs %}
+
 {% callout type="warning" title="Package Names with Multiple Slashes" %}
 The `package.json` name can only have one `/` character in it. This is more restrictive than the TypeScript path aliases. So if you have a project that you have been referencing with `@myorg/shared/ui`, you'll need to make the `package.json` name be something like `@myorg/shared-ui` and update all the import statements in your codebase to reference the new name.
 {% /callout %}
 
 ## Update Individual Project TypeScript Configuration
 
-Each project's `tsconfig.json` file should extend the `tsconfig.base.json` file and list `references` to the project's dependencies.
+Each project's `tsconfig.json` file should extend the `tsconfig.base.json` file and list `references` to the project's dependencies. Remove any `compilerOptions` listed and combine them with the options listed in the `tsconfig.lib.json` and `tsconfig.spec.json` files.
```

Comment:
as explained below, I am questioning this, as I think, the compileroptions are indeed merged with the extends option in a tsconfig, but not deepmerged. meaning: primitive entries will either be overwritten or merged with other entries, but nested records will be overwritten completely. (not sure what the intention is behind this instruction otherwise).

I did not verify this behavior, but it is my observation and I think my projects should not be able to compile if that was not the case, as I rely on this assumption for many files. Also I found a little backing for my claim in the second comment below this post: https://stackoverflow.com/a/71355500/3433137
"@cbdeveloper it merges compilerOptions -- you can see by running tsc --showconfig"